### PR TITLE
RHINENG-10603 first page table functionality - Pagination + Sorting

### PR DESCRIPTION
**Pull request by PreetiW**
## Hey, I just made a Pull Request!

Jira Ticket: https://issues.redhat.com/browse/RHINENG-10603

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: workspaces/resource-optimization/packages/app/src/App.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ import { CatalogGraphPage } from '@backstage/plugin-catalog-graph';
 import { RequirePermission } from '@backstage/plugin-permission-react';
 import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common/alpha';
 import { ResourceOptimizationPage } from '@backstage-community/plugin-resource-optimization';
+import { ResourceOptimizationDetailPage } from '@backstage-community/plugin-resource-optimization/src/plugin';
```
Comment (jkilzi) [inline workspaces/resource-optimization/packages/app/src/App.tsx]: Should be:
```suggestion
import { ResourceOptimizationDetailPage } from '@backstage-community/plugin-resource-optimization';
```

File: workspaces/resource-optimization/plugins/resource-optimization/src/utils/dates.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export const getTimeFromNow = (lastDate: string = 'N/A') => {
```
Comment (jkilzi) [inline workspaces/resource-optimization/plugins/resource-optimization/src/utils/dates.ts:1]: We can use a test for this module 😉

**Review by jkilzi — CHANGES_REQUESTED** [review]
Adjust the import in packages/app + write some unit tests for the `getTimeFromNow` function or maybe replace it with https://www.npmjs.com/package/react-moment

File: workspaces/resource-optimization/packages/app/src/App.tsx (hunk 1 above)
Comment (PreetiW) [reply workspaces/resource-optimization/packages/app/src/App.tsx]: good catch, fixed it 👍🏻

File: workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx [hunk 3]
```diff
@@ -39,93 +38,107 @@ const SELECT_ITEMS = [
...
-    items={SELECT_ITEMS}
-    multiple
-    onChange={() => {}}
-  />
+  <>
+    <SearchBar
+        placeholder="Filter by Project"
+        debounceTime={700}
+    />
+  </>
```
Comment (jkilzi) [inline workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx]:
```suggestion
    <SearchBar
        placeholder="Filter by Project"
        debounceTime={700}
    />
```

File: workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx [hunk 4]
```diff
@@ -39,93 +38,107 @@ const SELECT_ITEMS = [
...
 
 const WorkloadFilter = () => (
-  <Select
-    placeholder="All results"
-    label="WORKLOAD"
-    items={SELECT_ITEMS}
-    multiple
-    onChange={() => {}}
-  />
+  <>
```
Comment (jkilzi) [inline workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx]: Why these React.Fragments keep appearing where we don't need them? 😅

**Review by jkilzi — CHANGES_REQUESTED** [review]
PTAL at the remarks regarding the React.Fragments

File: workspaces/resource-optimization/plugins/resource-optimization/src/utils/dates.ts (hunk 2 above)
Comment (PreetiW) [reply workspaces/resource-optimization/plugins/resource-optimization/src/utils/dates.ts:1]: added the tests, will push the changes in next commit 💯

File: workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx (hunk 4 above)
Comment (PreetiW) [reply workspaces/resource-optimization/plugins/resource-optimization/src/components/ExampleComponent/ExampleComponent.tsx]: sorry didn't focus much on cleanup for now, I have removed these and will keep cleaning up the code in upcoming PRs.

**Review by jkilzi — APPROVED** [review]

**Pull request merged**